Makefile

```diff
@@ -10,8 +10,36 @@ LOCAL_IMAGE_REGISTRY ?= ko.local
 IMG_REPO=${LOCAL_IMAGE_REGISTRY}/cluster-api-provider-nutanix
 IMG_TAG=e2e-${GIT_COMMIT_HASH}
 MANAGER_IMAGE=${IMG_REPO}:${IMG_TAG}
-DOCKER_SOCKET := $(shell docker context inspect --format '{{.Endpoints.docker.Host}}')
-
+# Container engine detection (lazy evaluation - only computed when needed)
+CONTAINER_ENGINE = $(shell (command -v docker >/dev/null 2>&1 && echo docker) || \
+                           (command -v podman >/dev/null 2>&1 && echo podman) || \
+                           echo none)
+
+# Docker socket path (lazy evaluation)
+DOCKER_SOCKET = $(shell if [ "$(CONTAINER_ENGINE)" = "docker" ]; then \
+                           docker context inspect --format '{{.Endpoints.docker.Host}}'; \
+                        elif [ "$(CONTAINER_ENGINE)" = "podman" ]; then \
+                           echo "unix:///tmp/podman.sock"; \
```

Comment:
we shouldn't hardcode a socket in the makefile. This value could change setup to setup. Use a runtime command to identify the correct one.
```
podman machine inspect --format '{{ .ConnectionInfo.PodmanSocket.Path }}'
```